ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -324,7 +327,14 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			}
 		}
 	}
-	return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
+
+	if isJobPendingOrRunning(rayJobInstance.Status.JobStatus) {
```

Comment:
Is there a possibility that `JobStatus` is empty at this point, causing it to `return ctrl.Result{}, nil`?

Reply:
The only path, other than when the RayJob is in a terminal state, is the reconciliation cycle where the Kubernetes Job is created. However, now that this Job is watched, further reconciliation is guaranteed.

That being said, I've added an extra change in f7b639bf8bda205f739c6331d6b02fa0a093870c that initialises the Job status, instead of eventually have it updated once the ray job has started. This is generally a best practice, and it'll make sure periodic reconciliation will be triggered when `isJobPendingOrRunning` is true.

Reply:
Great!